### ISSUE TITLE
How to adjust audio latency?

### ISSUE THREAD

**Issue by Botspot** [ammen99/wf-recorder#264]
My application involves voice narration and a video feed of the webcam, inside a small mplayer window on-screen to be captured along with everything else.
The problem is that the webcam previewer has a bit of lag, causing the audio to play about 0.5s ahead of the video.

Is there a easily known way to tell ffmpeg to add latency to the audio? I would rather not do this with post-processing.

**Comment by soreau** [comment]
Are you using gstreamer for the webcam preview as mentioned in the wiki?

**Comment by Botspot** [comment]
> Are you using gstreamer for the webcam preview as mentioned in the wiki?

No, that seemed to perform very poorly, and the window could not be resized or moved.
I was using this:
```
mplayer -vo xv tv:// -tv driver=v4l2:device=/dev/video0:width=640:height=480:fps=30
```

**Comment by soreau** [comment]
Maybe try something like `mpv --gpu-context=wayland --hwdec=vaapi`.

**Comment by Botspot** [comment]
I've asked for help here: https://forums.raspberrypi.com/viewtopic.php?t=371908

**Comment by soreau** [comment]
For what it's worth, you should be able to resize the gst-launch window with the wayfire resize binding, the default binding is [here](https://github.com/WayfireWM/wayfire/blob/master/metadata/resize.xml#L10). (Super+Right Click Drag)

**Comment by Botspot** [comment]
This works extremely well.
```
mpv av://v4l2:/dev/video0 --profile=low-latency --untimed=yes --video-latency-hacks=yes --wayland-disable-vsync=yes --video-sync=display-desync
```
I would still like to delay the audio just slightly, so not closing this issue quite yet, but I suggest you add this to the wiki in case it helps others.

**Comment by Botspot** [comment]
Is there a way to pass arbitrary flags to ffmpeg? I think the `--itoffset` [flag](https://superuser.com/questions/538031/what-is-difference-between-ss-and-itsoffset-in-ffmpeg) does what I want.

Things I've tried:
```
yes | wf-recorder -a -f /tmp/video.mkv -c libx264 -p preset=ultrafast -r 30 -F=-itsoffset=5
#fails
yes | wf-recorder -a -f /tmp/video.mkv -c libx264 -p preset=ultrafast -r 30 -P=-itsoffset=5
#has no effect
```

**Comment by soreau** [comment]
I think you might want `--codec-param="-itsoffset=5"`

**Comment by Botspot** [comment]
> I think you might want `--codec-param="-itsoffset=5"`

No change. For ffmpeg, `-itsoffset` must be specified before the input you want to offset, in case that detail becomes important.

**Comment by Botspot** [comment]
Solution found in #294
